erizo/src/erizo/rtp/RtpExtensionProcessor.cpp

```diff
@@ -5,118 +5,94 @@
+                   extension.uri.c_str(),
+                   extension.value);
+        auto item = translationMap_.find(extension.uri);
+        if (item != translationMap_.end()) {
+          auto& translator =
+              extension.mediaType == AUDIO_TYPE ? audio_extension_id_translator_ : video_extension_id_translator_;
+          translator.addDestMapItem(std::make_pair(extension.value, item->second));
         }
-        break;
-      case AUDIO_TYPE:
-        if (isValidExtension(theMap.uri)) {
-          ELOG_DEBUG("Adding RTP Extension for Audio %s, value %u", theMap.uri.c_str(), theMap.value);
-          ext_map_audio_[theMap.value] = RTPExtensions((*translationMap_.find(theMap.uri)).second);
-        } else {
-          ELOG_WARN("Unsupported extension %s", theMap.uri.c_str());
-        }
-        break;
-      default:
-        ELOG_WARN("Unknown type of extMap, ignoring");
-        break;
+      } else {
+        ELOG_WARN("Unsupported extension %s", extension.uri.c_str());
+      }
+    } else {
+      ELOG_WARN("Unknown type of extMap, ignoring");
     }
   }
 }
 
-bool RtpExtensionProcessor::isValidExtension(std::string uri) {
+bool RtpExtensionProcessor::isValidExtension(std::string uri) const {
   auto value = std::find_if(ext_mappings_.begin(), ext_mappings_.end(), [uri](const ExtMap &extension) {
     return uri == extension.uri;
   });
   return value != ext_mappings_.end() && translationMap_.find(uri) != translationMap_.end();
 }
 
 uint32_t RtpExtensionProcessor::processRtpExtensions(std::shared_ptr<DataPacket> p) {
-  const RtpHeader* head = reinterpret_cast<const RtpHeader*>(p->data);
+  auto* head = reinterpret_cast<RtpHeader*>(p->data);
   uint32_t len = p->length;
-  std::array<RTPExtensions, 10> extMap;
-  if (head->getExtension()) {
-    switch (p->type) {
-      case VIDEO_PACKET:
-        extMap = ext_map_video_;
-        break;
-      case AUDIO_PACKET:
-        extMap = ext_map_audio_;
-        break;
-      default:
-        ELOG_WARN("Won't process RTP extensions for unknown type packets");
-        return 0;
-        break;
-    }
+
+  if ((p->type == VIDEO_PACKET || p->type == AUDIO_PACKET) &&
+      head->getExtension() && head->getExtId() == 0xBEDE) {
     uint16_t totalExtLength = head->getExtLength();
-    if (head->getExtId() == 0xBEDE) {
-      char* extBuffer = (char*)&head->extensions;  // NOLINT
-      uint8_t extByte = 0;
-      uint16_t currentPlace = 1;
-      uint8_t extId = 0;
-      uint8_t extLength = 0;
-      while (currentPlace < (totalExtLength*4)) {
-        extByte = (uint8_t)(*extBuffer);
+    char* extBuffer = reinterpret_cast<char*>(&head->extensions);
+    uint8_t extByte = 0;
+    uint16_t currentPlace = 1;
+    uint8_t extId = 0;
+    uint8_t extLength = 0;
+    while (currentPlace < (totalExtLength * 4)) {
+      extByte = (uint8_t) (*extBuffer);
+      if (extByte != RtpExtentionIdTranslator::padding) {
```

Comment:
RtpExtentionIdTranslator::unknown might be more descriptive in this case, am I right? When reading the PR I thought you were adding RTP Padding in some case

Reply:
Sure, RtpExtentionIdTranslator::unknown would be more clear in this case, I'll add it as static constexpr unknown = 0x00; (because extension 0 can't be a valid extension).